### PR TITLE
CHE-994 Update the list of machines on toolbar and console panel

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Guliy vguliy@codenvy.com

### CODE REVIEW

File: core/ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/dropdown/DropDownWidget.java [hunk 1]
```diff
@@ -39,7 +51,7 @@
     void selectElement(@Nullable String id, @Nullable String name);
 
     /**
-     * Create an instance of element action with given name amd id for displaying it.
+     * Creates an instance of element action with given name amd id for displaying it.
```
Comment (azatsarynnyy) [inline core/ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/dropdown/DropDownWidget.java]: typo - ... name **amd** id ...

Comment (vitaliy-guliy) [inline core/ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/dropdown/DropDownWidget.java]: Thanks

**Comment by vparfonov** [comment]
ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java [hunk 2]
```diff
@@ -124,9 +136,9 @@ public void apply(List<MachineDto> list) throws OperationException {
                     public void apply(List<RecipeDescriptor> recipeList) throws OperationException {
                         for (RecipeDescriptor recipe : recipeList) {
                             // Display only "ssh" recipes
-                            if (!"ssh".equalsIgnoreCase(recipe.getType())) {
-                                continue;
-                            }
+//                            if (!"ssh".equalsIgnoreCase(recipe.getType())) {
```
Comment (vzhukovs) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:139]: Do we need commented code here?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java [hunk 3]
```diff
@@ -574,29 +592,31 @@ private void onConnectingFailed() {
...
-        for (MachineDto machine : machines) {
-            if (machine.getConfig().getName().equals(selectedTarget.getName()) &&
-                    "ssh".equals(machine.getConfig().getType()) &&
-                    machine.getStatus() == MachineStatus.RUNNING) {
-                machineId = machine.getId();
-                break;
+        if (selectedTarget != null && selectedTarget.isConnected()) {
+            for (MachineDto machine : machines) {
+                if (machine.getConfig().getName().equals(selectedTarget.getName()) &&
+                        "ssh".equals(machine.getConfig().getType()) &&
```
Comment (vzhukovs) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:598]: I'd rather move `"ssh"` to the constant.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java (hunk 2 above)
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:139]: Yes, the code is commented temporary to display all recipes

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java (hunk 3 above)
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:598]: Let we leave the code as it is. I have calculated 15 string constants in the class and not sure we have to move them to constants.

**Pull request merged**